### ISSUE TITLE
Keras training dies when running more than 1 process simultaneously

### ISSUE THREAD

**Issue by btlk** [keras-team/keras#8017]
Hello everyone, this is a x-post of [StackOverflow issue](https://stackoverflow.com/questions/46446670/keras-training-dies-when-running-more-than-1-process-simultaneously), which didnt get much attention.

============================================

Below I state that I tried all combinations of Keras 2.0.3/2.0.6 and Tensorflow-gpu 1.2.1/1.3 - this is due to significant codebase which is broken after updating to TF 1.4 + Keras 2.0.8.  
Patchnotes didnt explicitly mention improvements regarding this issue, thus I suppose it should persist there.

============================================

While working on parallel training of an ensemble of identical models I encountered the following bug:  
If I run two (or more) training processes in separate terminals, eventually all of them except for one (without a particular order or timing) will start outputting lines like...  
 
    Epoch 46/10000
    0s - loss: 3.8659 - sig_eq: 0.5004
    Epoch 47/10000
    0s - loss: 3.1350 - sig_eq: 0.4060
    Epoch 48/10000
    0s - loss: 0.0000e+00 - sig_eq: 0.0000e+00
    Epoch 49/10000
    0s - loss: 0.0000e+00 - sig_eq: 0.0000e+00
    Epoch 50/10000
    0s - loss: 0.0000e+00 - sig_eq: 0.0000e+00
    Epoch 51/10000
    0s - loss: 0.0000e+00 - sig_eq: 0.0000e+00
    Epoch 52/10000
    0s - loss: 0.0000e+00 - sig_eq: 0.0000e+00

...and one of them will continue working like normal. I should stress that it **never** occurs when launching only 1 process.

Thanks to `nvidia-smi` I know that when it happens, **CUDA UNKNOWN_ERROR** rises, though, training appears to continue.

Originally, I encountered this behavior when I used custom multiprocessing queue/pool combination for ensemble training on multiple gpus, but it proved to be reproducible even on a single gpu multiprocessing and, later, even using multiple terminals is OS.

I have this issue on 2 different Linux PCs:

<pre>
1: Arch, Nvidia GTX 970, CUDA 8.0, CUDNN 5.1/6.0  
2: Ubuntu 16.04 LTS, Nvidia GTX 1080, CUDA 8.0, CUDNN 6.0  
</pre>

Both were tested using Keras 2.0.3/2.0.6 with Tensorflow-GPU 1.2.1/1.3 backend. 

The code snippet to reproduce this issue is below  

```python

import tensorflow as tf
import numpy as np
from keras.models import Model
from keras.layers import (Input, Dense, Conv2D, Flatten)
from keras.initializers import RandomNormal


def expg_loss(y_true, y_pred):
    abval = tf.abs(y_pred - y_true)
    num_values = np.prod(y_pred.get_shape().as_list()[1:])
    return tf.where(
        abval > 0.0, tf.exp(-1.0 / abval), tf.zeros_like(abval)
    ) * num_values


def sig_eq(y_true, y_pred):
    values = tf.cast(tf.equal(
        tf.cast(tf.sign(y_true), tf.float32),
        tf.cast(tf.sign(y_pred), tf.float32)
    ), tf.float32)
    return tf.reduce_mean(values)


if __name__ == '__main__':
    config = tf.ConfigProto()
    config.gpu_options.allow_growth = True
    with tf.Session(config=config).as_default():
        inputs = Input(shape = [1, 86, 1])

        conv1 = Conv2D(
            filters = 2,
            kernel_size = (1, 1),
            activation = 'tanh',
            use_bias = False,
            kernel_initializer = RandomNormal(stddev = 1.0),
            name = 'pre'
        )(inputs)

        flatten = Flatten()(conv1)

        fc1 = Dense(
            units = 11,
            activation = 'tanh',
            use_bias = False,
            kernel_initializer = RandomNormal(stddev = 0.5),
            name = 'final'
        )(flatten)

        model = Model(inputs = inputs, outputs = fc1)

        model.compile(
            loss = expg_loss,
            optimizer = 'sgd',
            metrics = [sig_eq]
        )

        X_train = np.random.uniform(size=[30000, 1, 86, 1], high=100)
        y_train = np.random.uniform(size=[30000, 11], low=-1.98, high=1.98)

        model.fit(
            x = X_train, 
            y = y_train,
            batch_size = 256,
            epochs = 10000,
            verbose = 2
        )
```

Might be worth noting the following:  

- The error is likely to be reproduced faster when starting new training processes after aborting old ones;  
- I couldn't encounter the error without using convolutional layers (cudnn, is it you little devil?);  
- I tried delayed startup of the second process (who knows, maybe simultaneous tensorflow initialization caused it) - the error persisted anyway;  
- I tried both python 2.7 and 3.6.  

I will be glad to any input as I kinda ran out of ideas.

**Comment by harshini-gadige** [comment]
Is this still an issue ?
